### PR TITLE
examples: update with new numalogic and pynumaflow

### PR DESCRIPTION
**Pull request by ab93**
- update examples for numalogic version and pynumaflow versions
- base numalogic udf
- class-based udfs
- remove protobuf 3.20 requirement

### CODE REVIEW

File: examples/numalogic-simple-pipeline/numa-pl.yaml
Comment (ab93) [inline examples/numalogic-simple-pipeline/numa-pl.yaml:1]: Ignore this file. Will change the image to a quay one.

File: examples/numalogic-simple-pipeline/src/utils.py [hunk 1]
```diff
@@ -16,36 +15,44 @@
...
+        """Converts the payload to json."""
+        return json.dumps(asdict(self)).encode("utf-8")
+
+    @classmethod
+    def from_json(cls, json_payload: bytes) -> Self:
+        """Converts the json payload to Payload object."""
+        return cls(**json.loads(json_payload.decode("utf-8")))
+
+
+class NumalogicUDF:
```
Comment (ab93) [inline examples/numalogic-simple-pipeline/src/utils.py:43]: We can move this to numalogic, if we all agree on the structure.

Comment (nkoppisetty) [reply examples/numalogic-simple-pipeline/src/utils.py:43]: this a good one

**Review by nkoppisetty — APPROVED** [review]
LGTM

**Review by vigith — APPROVED** [review]

**Pull request merged**